### PR TITLE
[WIP][Doctests] Fix rst -> mdx

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

In this PR, the already doc tests are enabled. The changes to `quickstart.mdx` show the necessary changes to make the doc tests work

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: .github/workflows/doctests.yml [hunk 1]
```diff
@@ -4,9 +4,16 @@ on:
   push:
     branches:
       - doctest*
-  repository_dispatch:
-  schedule:
-    - cron: "0 0 * * *"
+  pull_request:
```
Comment (patrickvonplaten) [inline .github/workflows/doctests.yml:7]: Will revert this once PR is approved

**Comment by patrickvonplaten** [comment]
@LysandreJik @sgugger, It'd be great if you could take a first look here.

There are essentially two things that I'd like to verify/discuss here:
1. The seperation line `===PT-TF-SPLIT===` in Python code ```python is interpreted as an output which then leads the doctest to fail with:
```
Expected: ===PT-TF-SPLIT===
Got: Nothing
```
In this PR I replaced `===PT-TF-SPLIT===` with `>>> # ===PT-TF-SPLIT===` so that the split is considered as a comment by doctests and they pass.

2. The second problem is that if the code snippet ends with ``` directly after the last python command, the doctests also interpret this as an expected output. This is why I added a newline before every ``` in the end which is not really a good idea IMO. I think it might make sense to just a quick preprocessing of all files that we run the doc tests on to add an empty space before the final ``` -> and then run the doc tests. Think this is better than adding a new line before every ``` which also doesn't go well with @sgugger's style_doc.py file.

File: docs/source/quicktour.mdx [hunk 2]
```diff
@@ -59,6 +59,7 @@ pip install tensorflow
 >>> from transformers import pipeline
```
Comment (patrickvonplaten) [inline docs/source/quicktour.mdx:59]: This file as changed right now would pass the doc-tests

File: docs/source/quicktour.mdx [hunk 3]
```diff
@@ -81,6 +84,7 @@ a list of dictionaries like this one:
 ...     print(f"label: {result['label']}, with score: {round(result['score'], 4)}")
 label: POSITIVE, with score: 0.9998
 label: NEGATIVE, with score: 0.5309
+
```
Comment (patrickvonplaten) [inline docs/source/quicktour.mdx:87]: This is what I meant with my 2. point

File: docs/source/quicktour.mdx [hunk 4]
```diff
@@ -113,8 +118,9 @@ We will need two classes for this. The first is [`AutoTokenizer`], which we will
 
 ```py
 >>> from transformers import AutoTokenizer, AutoModelForSequenceClassification
-===PT-TF-SPLIT===
+>>> # ===PT-TF-SPLIT===
```
Comment (patrickvonplaten) [inline docs/source/quicktour.mdx:121]: This is what I meant with my 1. point

File: utils/documentation_tests.txt [hunk 5]
```diff
@@ -1,2 +1,2 @@
-docs/source/quicktour.rst
-docs/source/task_summary.rst
\ No newline at end of file
+docs/source/quicktour.mdx
```
Comment (patrickvonplaten) [inline utils/documentation_tests.txt:1]: this one would pass as is now

File: utils/documentation_tests.txt [hunk 6]
```diff
@@ -1,2 +1,2 @@
...
+docs/source/task_summary.mdx
```
Comment (patrickvonplaten) [inline utils/documentation_tests.txt:2]: this one doesn't pass as is now

**Review by LysandreJik — COMMENTED** [review]
Sorry I got a bit lost, do you still need a review on this one @patrickvonplaten?

**Comment by patrickvonplaten** [comment]
Closing this one as it's more or less a duplicate of the one we merged last week and requires the `PT-TF` switch to be solved before. Will open a new one then :-)

**Pull request closed without merging**